### ISSUE TITLE
Module compiled with Swift 5.0 cannot be imported by the Swift 5.1 compiler

### ISSUE THREAD

**Issue by pobyt** [razorpay/razorpay-flutter#28]
iOS build is failing with below exception with new XCODE release Version 11.0.
Please update your iOS SDK to 5.1 in Flutter Project.
Module compiled with Swift 5.0 cannot be imported by the Swift 5.1 compiler: /ios/Pods/razorpay-pod/Pod/Razorpay.framework/Modules/Razorpay.swiftmodule/x86_64.swiftmodule

**Comment by muthufmass** [comment]
any update on this, We are also blocked on the same error. Needs a rebuild and new version with latest xcode compiler on mac.

**Comment by nihalreddy33** [comment]
@ChintanAcharya any update/ETA on this issue?

**Comment by abhikumar08** [comment]
facing the same issue, did anyone find a way around to get this work?

**Comment by prasanna-narayanan24** [comment]
https://razorpay.com/docs/payment-gateway/ios-integration/standard/

A temporary fix is, Download the latest 5.1 razorpay-sdk (Razorpay.framework). Replace this downloaded one in ios/Pods/razorpay-pod/Pod/Razorpay.framework.

**Comment by nihalreddy33** [comment]
@prasanna-narayanan24 if we do that we are getting below exception when we run.
/flutter/.pub-cache/hosted/pub.dartlang.org/razorpay_flutter-1.1.1/ios/Classes/RazorpayDelegate.swift:17:17: Method 'onExternalWalletSelected(_:WithPaymentData:)' has different argument labels from those required by protocol 'ExternalWalletSelectionProtocol' ('onExternalWalletSelected(_:withPaymentData:)')

**Comment by prasanna-narayanan24** [comment]
> @prasanna-narayanan24 if we do that we are getting below exception when we run.
> /flutter/.pub-cache/hosted/pub.dartlang.org/razorpay_flutter-1.1.1/ios/Classes/RazorpayDelegate.swift:17:17: Method 'onExternalWalletSelected(_:WithPaymentData:)' has different argument labels from those required by protocol 'ExternalWalletSelectionProtocol' ('onExternalWalletSelected(_:withPaymentData:)')

can you delete the Podfile.lock file and try doing the same steps as mentioned above?

**Comment by nihalreddy33** [comment]
@prasanna-narayanan24 even after deleting the podfile.lock, I am getting the same exception with the new 5.1 SDK.

**Comment by prasanna-narayanan24** [comment]
> @prasanna-narayanan24 even after deleting the podfile.lock, I am getting the same exception with the new 5.1 SDK.

the error says that withPaymentData is replaced with WithPaymentData, so i guess changing the variable name should fix.

**Comment by prasanna-narayanan24** [comment]
```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      **config.build_settings['SWIFT_VERSION'] = '5.1'**
    end
  end
end
```

Also fixes

**Comment by ChintanAcharya** [comment]
We are working on a Swift 5.1 compatible release. ETA is by the end of this week.

**Comment by nihalreddy33** [comment]
Thank you @ChintanAcharya  for the update.

**Comment by ChintanAcharya** [comment]
v1.1.2 is out, which should fix this issue.

**Comment by meera13raveendran** [comment]
> v1.1.2 is out, which should fix this issue.

hi..just updating the pod file and updating razorpay pod will work..or is there anything else I should do?

**Comment by LOG-TAG** [comment]
what is the version of pod that is compatible on xcode 10.1 ?
<img width="565" alt="Screen Shot 2019-10-20 at 12 50 13 PM" src="https://user-images.githubusercontent.com/1190344/67156148-7b11e800-f338-11e9-8403-191311e91469.png">

**Comment by LOG-TAG** [comment]
@prasanna-narayanan24 any workaround possible for xcode 10.1  ? alfa version of pub will work?

**Comment by meera13raveendran** [comment]
Hi
Thanks for the reply.. I did update the razorpay pod but xcode didn't
reflect the change.. But next day when xcode launched error was gone.. .
Issue fixed hence..

Once again  thanks for the reply..

**Comment by MustansirZia** [comment]
Updating to v1.1.2 fixed this issue for us. Thanks!

**Comment by meera13raveendran** [comment]
Yes I did update the pod and working fine now.. Thanks for the reply
1. Edit the podfile 
'pod razorpay-pod in podfile '
2. And in terminal run the command pod install /
Pod update razorpay-pod 
worked for me

**Comment by meera13raveendran** [comment]
Yes I did update the pod and working fine now.. Thanks for the reply
1. Edit the podfile 
'pod razorpay-pod in podfile '
2. And in terminal run the command pod install /
Pod update razorpay-pod 
worked for me

**Comment by Nautiyalsachin** [comment]
Closing this issue due to the time elapsed. Please feel free to either re-open the issue or create a new ticket if you have any additional issues.

**Comment by deeppanchal** [comment]
> Updating to v1.1.2 fixed this issue for us. Thanks!

Which Xcode Version you are using?

**Comment by vikram94** [comment]
Here is how I resolved the issue.. Hope helps someone and saves his time.

Put in Podfile - pod 'razorpay-pod', '1.1.5'
Delete Pod Folder
Put 'use_frameworks!' Below target ‘Runner’ do
Type Command in iOS - Pod update razorpay-pod
flutter pub pub cache repair
Flutter build iOS

in case of an issue do ask

**Comment by balasubramanian1612s** [comment]
> issue

Could you please explain a bit!?

**Comment by deepakatom** [comment]
Got the issue again with Swift 5.2.4 this week. Any solution for this?
![error](https://user-images.githubusercontent.com/71891121/94283224-5c679880-ff6e-11ea-91c9-ed364a592a3e.jpeg)

**Comment by Nautiyalsachin** [comment]
Hey @deepakatom, Can you try cleaning the build and deleting derived data first, and try again, let me know after following these steps you are still getting the same error.
